improve rest api documentation: first draft (#3306)
status-im/nim-beacon-chain: docs/the_nimbus_book/src/rest-api.md

# REST API

> Nimbus exposes an **extremely fast** and high-performance implementation of the standard [Beacon Node API](https://ethereum.github.io/beacon-APIs/). The goal of a standard API is to improve interoperability between various beacon node implementations.

Reviewer: ```suggestion
Nimbus exposes an **extremely fast** and high-performance implementation of the standard [Beacon Node API](https://ethereum.github.io/beacon-APIs/). The API allows you to use Nimbus together with third-party tooling such as validator clients, block explorers as well as your own monitoring infrastructure.

> The API is a `REST` interface accessed via `HTTP`. **The API should not**, unless protected by additional security layers, **be exposed to the public Internet** as the API includes multiple endpoints which could open your node to denial-of-service (DoS) attacks.

Reviewer: ```suggestion
The API is a `REST` interface accessed via `HTTP`. **The API should not be exposed to the public Internet** unless protected by additional security: it includes multiple endpoints which could open your node to denial-of-service (DoS) attacks.
```

The API can be used with any conforming consumer, including alternative validator client implementations, explorers and tooling.

## Test your tooling against our servers

 The API is available from:

* `http://testing.mainnet.beacon-api.nimbus.team/`
* `http://unstable.mainnet.beacon-api.nimbus.team/`
* `http://unstable.prater.beacon-api.nimbus.team/`

You can make requests as follows (here we are requesting the version the Nimbus software version of the node in question):

#### Mainnet testing branch
```
curl -X GET http://testing.mainnet.beacon-api.nimbus.team/eth/v1/node/version
```

#### Mainnet unstable branch
```
curl -X GET http://unstable.mainnet.beacon-api.nimbus.team/eth/v1/node/version
```

#### Prater unstable branch
```
curl -X GET  http://unstable.prater.beacon-api.nimbus.team/eth/v1/node/version
```

> Note that right now these are very much unstable testing instances. They may be unresponsive at times - so **please do not rely on them for validation**. We may also disable them at any time without warning.

Reviewer: ```suggestion
The test endpoints are part of pre-release testing and run an unstable version of Nimbus - we welcome reports about any problems you might have with them.

They may also be unresponsive at times - so **please do not rely on them for validation**. We may also disable them at any time without warning.
```

## Configure your node to run a local REST server

> > **Warning:** If you are using a validator client with a Nimbus beacon node, and running a Nimbus version prior to `v1.5.5`,  then you will need to launch the node with the `--subscribe-all-subnets` option enabled (in addition to the `--rest` option).

Reviewer: let's move this comment to the end of the section, it becomes less and less relevant with every release

By default, the REST interface is disabled. To enable it, start the beacon node with the `--rest` option:

```

> ./run-prater-beacon-node.sh --rest

Reviewer: we have a mix of mainnet / prater throughout the manual - we should settle on one I think (mainnet probably).

Reviewer: this helps establish 5052 as the "mainnet beacon node port" - then we can use 5152 (or something) in examples that involve testnets

Author: sounds good
